### PR TITLE
feat: useAB with SWR (no Context or Provider)

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Leverages SWR to provide the `ABTestAPI` via `useAB`. Instead of using a `Provider`, we have an Island dedicated to mutating the SWR cache (`SetABTests`).

This introduces a new pattern: [SWR’s `mutate`](https://swr.vercel.app/docs/mutation), which is an a tool usually built for optimistic data mutation, where we don’t wait for the promise to resolve. In this specific case, the promise itself will never resolve, and we have to set the value ourselves.

## Why?

We currently need to have AB Providers for each Island that relies on the AB Test API. This leads to duplication of props and a lot more boilerplate code.

We want the platform to be as easy as possible to use.

See #3955

### CODE REVIEW

**Comment by mxdvl** [comment]
This might not be fully baked yet… @OllysCoding feel free to pick up and modify as necessary.

File: dotcom-rendering/src/web/components/WithABProvider.tsx [hunk 1]
```diff
@@ -1,70 +0,0 @@
...
-	// ** Setup AB Test Tracking *****
-	// *******************************
-	const ABTestAPI = useAB();
-	useEffect(() => {
-		const allRunnableTests = ABTestAPI.allRunnableTests(tests);
-		ABTestAPI.trackABTests(allRunnableTests);
-		ABTestAPI.registerImpressionEvents(allRunnableTests);
-		ABTestAPI.registerCompleteEvents(allRunnableTests);
-		log('dotcom', 'AB tests initialised');
-	}, [ABTestAPI]);
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/WithABProvider.tsx:28]: ⚠️ This is missing in the new implementation.

**Review by oliverlloyd — APPROVED** [review]
I made two comments but we've already discussed this in person and it's all good 👍

File: dotcom-rendering/src/web/components/Page.tsx [hunk 2]
```diff
@@ -77,6 +71,13 @@ export const Page = ({ CAPI, NAV, format }: Props) => {
 					shouldHideReaderRevenue={CAPI.shouldHideReaderRevenue}
 				/>
 			</Island>
+			<Island clientOnly={true}>
```
[the remark this answers is not in the document]
Comment (mxdvl) [reply dotcom-rendering/src/web/components/Page.tsx:74]: I think this is down to consumer to make this decision. There’s a chance that the `ABTestApi` is undefined. Most cases are already using a `useOnce`, so it looks safe.

[the remark this answers is not in the document]
Comment (oliverlloyd) [reply dotcom-rendering/src/web/components/Page.tsx:74]: Sounds good 👍

**Pull request merged**